Product/Production/Services/DirectCore/src/main/java/gov/hhs/fha/nhinc/direct/messagemonitoring/impl/MessageMonitoringAPI.java

```diff
@@ -245,8 +249,10 @@ private void buildCache() {
         clearCache();
         //load the pending outgoing messages to the cache
         for (MonitoredMessage trackMessage : pendingMessages) {
-            messageMonitoringCache.put(trackMessage.getMessageid(), trackMessage);
-            LOG.debug("Total child rows for the messageId:" + trackMessage.getMonitoredmessagenotifications().size());
+            if (!trackMessage.getStatus().equals(STATUS_ARCHIVED)) {
```

Comment:
Its better to cleanup the Archived message on a server restart. Call deleteElapsedArchivedMessage() method if we find any archived message.